File: TourList.Model/Tour.cs
using System;
using System.Collections.Generic;

namespace TourList.Model
{
  public class Tour
  {
    public Guid Id { get; set; }
    public DateTime Date { get; set; }

    public Guid ExcursionId { get; set; }
    public Excursion Excursion { get; set; }

    public Guid ClientId { get; set; }
    public Client Client { get; set; }

    public IEnumerable<SnapshotSight> SnapshotSights { get; set; }

    public Tour()
    {
      SnapshotSights = new List<SnapshotSight>();
    }
  }
}

File: TourList-FrontEnd/src/app/dashboard/shared/models/tour.ts
import { Client } from './client';
import { Excursion } from './excursion';
import { ExcursionSight } from './excursion-sight';

export class Tour {
    id?: string;
    date: string;
    client: Client;
    excursion: Excursion;
    snapshotSights: ExcursionSight[];
}